File: Python/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    def __init__(self, plate_number, brand, year, color, type_motor, tank_capacity):
        # Konstruktor kelas Motorcycle, mewarisi atribut dari kelas Vehicle
        super().__init__(plate_number, brand, year, color)
        self.__type_motor = type_motor        # Inisialisasi jenis motor
        self.__tank_capacity = tank_capacity  # Inisialisasi kapasitas tangki bahan bakar

    def get_type_motor(self):
        # Metode untuk mendapatkan jenis motor
        return self.__type_motor

    def set_type_motor(self, type_motor):
        # Metode untuk mengatur jenis motor
        self.__type_motor = type_motor

    def get_tank_capacity(self):
        # Metode untuk mendapatkan kapasitas tangki bahan bakar
        return self.__tank_capacity

    def set_tank_capacity(self, tank_capacity):
        # Metode untuk mengatur kapasitas tangki bahan bakar
        self.__tank_capacity = tank_capacity

File: cpp2/Motorcycle.cpp
#include <iostream>
#include <string>
#include "Vehicle.cpp" // Include header file Vehicle.h

class Motorcycle : public Vehicle { // Menurunkan kelas Motorcycle dari kelas Vehicle
private:
    std::string type_motor;
    int tank_capacity;

public:
    // Konstruktor untuk kelas Motorcycle
    Motorcycle(std::string plate_number, std::string brand, int year, std::string color, std::string type_motor, int tank_capacity)
        : Vehicle(plate_number, brand, year, color), type_motor(type_motor), tank_capacity(tank_capacity) {}

    // Getter dan setter untuk jenis motor
    std::string get_type_motor() {
        return type_motor;
    }

    void set_type_motor(std::string type_motor) {
        this->type_motor = type_motor;
    }

    // Getter dan setter untuk kapasitas tangki
    int get_tank_capacity() {
        return tank_capacity;
    }

    void set_tank_capacity(int tank_capacity) {
        this->tank_capacity = tank_capacity;
    }
};